src/include/sof/list.h

```diff
@@ -75,8 +75,13 @@ static inline void list_item_append(struct list_item *item,
 /* delete item from the list */
 static inline void list_item_del(struct list_item *item)
 {
-	item->next->prev = item->prev;
-	item->prev->next = item->next;
+	if (item->next)
```

Comment:
Lets have this as a debug option. i.e. define a macro that enables setting next, prev = 0 and then does a check. This way we can use this on any list when we enable list debugging.